# Clayton Seabaugh: Midterm Project
**Author:** Clayton Seabaugh  
**Date:** 3-6-2025  
**Objective:** This project demonstrates my ability to apply classification modeling techniques to a real-world dataset. 

## Section 1: Import and Inspect the Data

### Section 1.1 Import modules and inspect first 10 rows

In [9]:
# Imports needed
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt
import os

In [ ]:
# Create file path for data and print head
file_path = 'data\data_banknote_authentication.txt'
if os.path.exists(file_path):
    # Read and print df
    df = pd.read_csv(file_path)
    print(df.head(10))
else:
    print(f"File {file_path} does not exist.")



   variance  skewness  curtiosis  entropy  class
0   3.62160    8.6661   -2.80730 -0.44699      0
1   4.54590    8.1674   -2.45860 -1.46210      0
2   3.86600   -2.6383    1.92420  0.10645      0
3   3.45660    9.5228   -4.01120 -3.59440      0
4   0.32924   -4.4552    4.57180 -0.98880      0
5   4.36840    9.6718   -3.96060 -3.16250      0
6   3.59120    3.0129    0.72888  0.56421      0
7   2.09220   -6.8100    8.46360 -0.60216      0
8   3.20320    5.7588   -0.75345 -0.61251      0
9   1.53560    9.1772   -2.27180 -0.73535      0


<>:2: SyntaxWarning: invalid escape sequence '\d'
<>:2: SyntaxWarning: invalid escape sequence '\d'
C:\Users\clayt\AppData\Local\Temp\ipykernel_15808\4251896938.py:2: SyntaxWarning: invalid escape sequence '\d'
  file_path = 'data\data_banknote_authentication.txt'


### Section 1.2 Check for missing values and display summary statitistics

In [35]:
# Check for missing values
missing_values = df.isnull().sum()
print(f"Missing Values in each Column:\n{missing_values}")

# Generate summary statistics
summary_stats = df.describe()
print (f"Summary Statistics:\n{summary_stats}")

Missing Values in each Column:
variance     0
skewness     0
curtiosis    0
entropy      0
class        0
dtype: int64
Summary Statistics:
          variance     skewness    curtiosis      entropy        class
count  1372.000000  1372.000000  1372.000000  1372.000000  1372.000000
mean      0.433735     1.922353     1.397627    -1.191657     0.444606
std       2.842763     5.869047     4.310030     2.101013     0.497103
min      -7.042100   -13.773100    -5.286100    -8.548200     0.000000
25%      -1.773000    -1.708200    -1.574975    -2.413450     0.000000
50%       0.496180     2.319650     0.616630    -0.586650     0.000000
75%       2.821475     6.814625     3.179250     0.394810     1.000000
max       6.824800    12.951600    17.927400     2.449500     1.000000


### Reflection 1:
- What do you notice about the dataset? Are there any data issues? I notice that there are no null values and that the dataset is primarily float numbers except class which are integers. I can see it is a rather large dataset with 1372 rows. 